test/widgets/autocomplete_test.dart

```diff
@@ -72,6 +74,52 @@ Future<Finder> setupToComposeInput(WidgetTester tester, {
   return finder;
 }
 
+/// Simulates loading a [MessageListPage] with a stream narrow
+/// and tapping to focus the topic input.
+///
+/// Also prepares test-topics to be sent to topics api request,
+/// so they can show up in autocomplete.
+Future<Finder> setupToTopicInput(WidgetTester tester, {
+  required List<Topic> topics,
+}) async {
+  addTearDown(testBinding.reset);
+  await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+  final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+  final connection = store.connection as FakeApiConnection;
+
+  // prepare message list data
+  final stream = eg.stream();
+  final message = eg.streamMessage(stream: stream, sender: eg.selfUser);
+  connection.prepare(json: GetMessagesResult(
+    anchor: message.id,
+    foundNewest: true,
+    foundOldest: true,
+    foundAnchor: true,
+    historyLimited: false,
+    messages: [message],
+  ).toJson());
+
+  await tester.pumpWidget(Builder(builder: (context) =>
+    MaterialApp(
+      theme: zulipThemeData(context),
+      localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+      supportedLocales: ZulipLocalizations.supportedLocales,
+      home: GlobalStoreWidget(
+        child: PerAccountStoreWidget(
+          accountId: eg.selfAccount.id,
+          child: MessageListPage(
+            narrow: StreamNarrow(stream.streamId)))))));
```

Comment:
A bit shorter, and also a bit more representative:

```suggestion
  await tester.pumpWidget(const ZulipApp());
  await tester.pump();
  final navigator = await ZulipApp.navigator;
  navigator.push(MessageListPage.buildRoute(accountId: eg.selfAccount.id,
    narrow: StreamNarrow(stream.streamId)));
```

I'd like to do this with `setupToComposeInput` too but haven't gotten around to it yet.